### PR TITLE
Adds healthcheck api

### PR DESCRIPTION
**Pull request by ashraffouda**
Signed-off-by: Ashraf Fouda <ashraf.m.fouda@gmail.com>

### CODE REVIEW

File: api/healthcheck/HealthCheck.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"gopkg.in/validator.v2"
+)
+
+// Node node in the
+type HealthCheck struct {
+  Id        string   `json:"id" validate:"nonzero"`
+	Name      string   `json:"name" validate:"nonzero"`
+	Resource  string   `json:"resource" validate:"nonzero"`
+	Status    string   `json:"status" validate:"nonzero"`
```
Comment (khaledkbadr) [inline api/healthcheck/HealthCheck.go]: Format please

File: api/healthcheck/health_api_Node.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"net/http"
+  "github.com/gorilla/mux"
+	"github.com/zero-os/0-orchestrator/api/tools"
+)
+
+// ListNodeHealth is the handler for GET /health/nodes
+// List NodeHealth
+func (api HealthCheckApi) ListNodeHealth(w http.ResponseWriter, r *http.Request) {
+  aysClient := tools.GetAysConnection(r, api)
+	vars := mux.Vars(r)
```
Comment (khaledkbadr) [inline api/healthcheck/health_api_Node.go]: format please

File: api/healthcheck/HealthCheck.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  Message   string   `json:"message" validate:"nonzero"`
+}
+
+type NodeHealthCheck struct {
+	Hostname  string  `json:"hostname" validate:"nonzero"`
+	Id        string  `json:"id" validate:"nonzero"`
```
Comment (khaledkbadr) [inline api/healthcheck/HealthCheck.go]: Change `Id` to `ID`

File: api/healthcheck/HealthCheck.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	Status    string  `json:"status" validate:"nonzero"`
+}
+
+
+type Node struct {
+	Hostname  string                `json:"hostname" validate:"nonzero"`
+	Id        string                `json:"id" validate:"nonzero"`
```
Comment (khaledkbadr) [inline api/healthcheck/HealthCheck.go]: Change Id to ID

File: api/healthcheck/health_api_Node.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package healthcheck
+
+import (
+	"encoding/json"
+	"net/http"
+  "github.com/gorilla/mux"
+	"github.com/zero-os/0-orchestrator/api/tools"
+)
+
+// ListNodeHealth is the handler for GET /health/nodes
```
Comment (khaledkbadr) [inline api/healthcheck/health_api_Node.go]: Change comment to node usage

File: api/healthcheck/health_api_Nodes.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	var respBody = make([]NodeHealthCheck, len(services))
+	for i, service := range services {
+		var node Node
+		if err := json.Unmarshal(service.Data, &node); err != nil {
+			tools.WriteError(w, http.StatusInternalServerError, err, "Error unmrshaling ays response")
+			return
+		}
+		fmt.Printf(node.Id)
```
Comment (khaledkbadr) [inline api/healthcheck/health_api_Nodes.go]: remove unnecessary print

File: api/healthcheck/health_api_Nodes.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+		respBody[i].Id = service.Name
+		respBody[i].Hostname = node.Hostname
```
Comment (khaledkbadr) [inline api/healthcheck/health_api_Nodes.go]: use append instead using index to fill the slice

File: api/healthcheck/health_api_Nodes.go [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+		healthstatus := "OK"
+		for _, healthcheck := range node.HealthChecks {
+			fmt.Printf(healthcheck.Status)
```
Comment (khaledkbadr) [inline api/healthcheck/health_api_Nodes.go]: remove unnecessary print

File: api/healthcheck/healthcheck_api.go [hunk 9]
```diff
@@ -0,0 +1,29 @@
+package healthcheck
+
+import (
+	ays "github.com/zero-os/0-orchestrator/api/ays-client"
+	_ "github.com/zero-os/0-orchestrator/api/validators"
+)
+
+// VdisksAPI is API implementation of /vdisks root endpoint
```
Comment (khaledkbadr) [inline api/healthcheck/healthcheck_api.go]: Change comment

File: api/healthcheck/healthcheck_api.go [hunk 10]
```diff
@@ -0,0 +1,29 @@
+package healthcheck
+
+import (
+	ays "github.com/zero-os/0-orchestrator/api/ays-client"
+	_ "github.com/zero-os/0-orchestrator/api/validators"
```
Comment (khaledkbadr) [inline api/healthcheck/healthcheck_api.go]: Check if this import is needed

File: api/healthcheck/healthcheck_if.go [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+//This file is auto-generated by go-raml
+//Do not edit this file by hand since it will be overwritten during the next generation
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+// Healthcheck Interface is interface for /vdisks root endpoint
```
Comment (khaledkbadr) [inline api/healthcheck/healthcheck_if.go]: Change comment

File: api/healthcheck/health_api_Nodes.go (hunk 7 above)
Comment (ashraffouda) [inline api/healthcheck/health_api_Nodes.go]: how?

File: raml/api.raml [hunk 12]
```diff
@@ -49,38 +73,6 @@ types:
           body:
             application/json:
               type: Node
```
Comment (grimpy) [inline raml/api.raml]: why was the stats section removed? Bad conflict resolvement?

**Pull request closed without merging**